index.js

```diff
@@ -61,6 +75,30 @@ export default function rehypeExternalLinks(options = {}) {
         const url = node.properties.href
         const protocol = url.slice(0, url.indexOf(':'))
 
+        const options_ =
+          typeof options === 'function' ? options(node, tree) : options
+        const exclude = options_.exclude || []
+
+        // Rehype camelizes some properties and doesn't for some others, to gurantee compatibility
+        // and avoid edge cases, I'm camelizing everything manually,
+        const propertyKeys = new Set(
+          Object.keys(node.properties).map((x) => camelCase(x))
+        )
```

Comment:
```suggestion
```

Hmm. This isn’t okay. It is very important to not change the casing. Users must provide the correct casing. This is intentional and documented in https://github.com/syntax-tree/hast#propertyname

Reply:
data properties are automatically camelized by rehype, that's why this solution is required. The exclude option won't work properly without either de-camelizing data properties, or camelizing all properties

Reply:
This is not a bug.
Known HTML properties are cases. Unknown properties are not.
`exclude-link` is not supposed to be there in HTML.
I recommend using a data property: `data-exclude-link`.
These are camelcased as `dataExcludeLink`.

Reply:
Ok